txmain/main.c

```diff
@@ -334,14 +352,24 @@ static void status_update(bool have_link)
         played_tone = true;
     }
     
-    if (!played_tone && (t_status.flags & TELEM_FLAG_VIDEO)) {
-        video_tone_counter++;
-        if (video_tone_counter == 2) {
-            video_tone_counter = 0;
-            buzzer_tune(TONE_VIDEO);
-            played_tone = true;
+    // Check for photo/video tone
+    if (t_status.flags & TELEM_FLAG_VIDEO)
+    {
+        if (!played_tone) {
+            video_tone_counter++;
+            if (video_tone_counter == 10) { // Beep once per second if it is held
+                video_tone_counter = 0;
+            }
+            if (video_tone_counter == 2) { // require at least two telemetry packets (100ms apart) saying video mode is enabled
+                buzzer_tune(TONE_VIDEO);
+                played_tone = true;
```

Comment:
Shouldn't this is have a 
video_tone_counter = 0;  since we just played the tone?

Reply:
It includes two copies of that line:
- After 1000ms it resets to 0, so that the beep is not too frequent, and
- When the current telemetry status flags do not include TELEM_FLAG_VIDEO (which the drone sets high for 1 second after the user takes a photo, or while a video is recording).
As per the comment, it will not beep if an isolated packet has TELEM_FLAG_VIDEO set; it requires it to be set in two subsequent calls (i.e. for 100...200ms)

Reply:
ok I had to write it out in english to get it.
as a protection, though it shouldn't happen, if you expect (for exmaple) at least 2 packets) then it should be >= rather than ==

I unfortunately have never been able to successfully compile the STM8TX code..so all i can do is review..:\

Reply:
You can compile it as is using SDCC under Linux.
I can compile it under SDCC on Windows or IAR on Windows (Beken branch), but that requires extra tools.

If we use >= instead of == then it would beep nearly every 100ms instead of every 1000ms, which would be annoying to the user.